crates/project/src/direnv.rs

```diff
@@ -0,0 +1,45 @@
+#[cfg(not(any(test, feature = "test-support")))]
+use {collections::HashMap, std::path::Path, util::ResultExt};
+
+#[derive(Clone)]
+pub struct DirenvError;
+
+#[cfg(not(any(test, feature = "test-support")))]
+pub async fn load_direnv_environment(
+    dir: &Path,
+) -> (Option<HashMap<String, String>>, Option<DirenvError>) {
```

Comment:
Can't this be a `Result<HashMap<String, String>>`? Optional value and optional result makes me think that this is exactly what `Result` is for :)

Reply:
No, because there could be cases like `(Some, Some)` or `(None, None)`

Reply:
Upon further consideration, implemented